File: frontend/src/commands/environment-cookie.ts
import { invoke } from './index'

export const environment_cookie_query_environment_uuid = async (environmentUuid: string) => {
    return await invoke('environment_cookie_query_environment_uuid', { environmentUuid })
}

export const environment_cookie_create = async (
    environmentUuid: string,
    cookie_str: string,
) => {
    return await invoke('environment_cookie_create', {
        environmentUuid,
        cookie_str
    })
}

export const environment_account_create = async (
    environmentUuid: string,
    cookie_str: string,
) => {
    return await invoke('environment_account_create', {
        environmentUuid,
        cookie_str
    })
}

export const environment_account_modify = async (
    environmentUuid: string,
    cookie_str: string,
) => {
    return await invoke('environment_account_modify', {
        environmentUuid,
        cookie_str
    })
}

export const environment_account_delete = async (environmentUuid: string) => {
    return await invoke('environment_account_delete', { environmentUuid })
}

File: crates/lp-views/src/command/environment_cookie.rs
use lp_cores::requests::JsonRespnse;
use lp_services_remote::requests::environment_cookie;

#[tauri::command]
pub async fn environment_cookie_query_environment_uuid(
    environment_uuid: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(environment_cookie::query_by_uuid(&environment_uuid).await?)
}

#[tauri::command]
pub async fn environment_cookie_create(
    environment_uuid: String,
    cookie_str: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(environment_cookie::create(&environment_uuid, &cookie_str).await?)
}

#[tauri::command]
pub async fn environment_cookie_modify(
    environment_uuid: String,
    cookie_str: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(environment_cookie::modify(&environment_uuid, &cookie_str).await?)
}

#[tauri::command]
pub async fn environment_cookie_delete(
    environmnet_uuid: &str,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(environment_cookie::delete(&environmnet_uuid).await?)
}
